examples/todo/modules/data_l1/src/main/scala/com/my/data_l1/DataL1Service.scala

```diff
@@ -0,0 +1,113 @@
+
+        override def serializeUpdate(update: TodoUpdate): F[Array[Byte]] =
+          JsonSerializer[F].serialize[TodoUpdate](update)
+
+        override def deserializeUpdate(bytes: Array[Byte]): F[Either[Throwable, TodoUpdate]] =
+          JsonSerializer[F].deserialize[TodoUpdate](bytes)
+
+        override def serializeBlock(block: Signed[DataApplicationBlock]): F[Array[Byte]] =
+          JsonSerializer[F].serialize[Signed[DataApplicationBlock]](block)
+
+        override def deserializeBlock(bytes: Array[Byte]): F[Either[Throwable, Signed[DataApplicationBlock]]] =
+          JsonSerializer[F].deserialize[Signed[DataApplicationBlock]](bytes)
+
+        override def serializeCalculatedState(calculatedState: CalculatedState): F[Array[Byte]] =
+          JsonSerializer[F].serialize[CalculatedState](calculatedState)
+
+        override def deserializeCalculatedState(bytes: Array[Byte]): F[Either[Throwable, CalculatedState]] =
+          JsonSerializer[F].deserialize[CalculatedState](bytes)
+
+        override def dataEncoder: Encoder[TodoUpdate] = implicitly(Encoder[TodoUpdate])
+
+        override def dataDecoder: Decoder[TodoUpdate] = implicitly(Decoder[TodoUpdate])
+
+        override def calculatedStateEncoder: Encoder[CalculatedState] = implicitly(Encoder[CalculatedState])
+
+        override def calculatedStateDecoder: Decoder[CalculatedState] = implicitly(Decoder[CalculatedState])
+
+        override val signedDataEntityDecoder: EntityDecoder[F, Signed[TodoUpdate]] = circeEntityDecoder
+
+        override def getCalculatedState(implicit
+          context: L1NodeContext[F]
+        ): F[(SnapshotOrdinal, CalculatedState)] =
+          (SnapshotOrdinal.MinValue, CalculatedState.genesis).pure[F]
+
+        override def setCalculatedState(ordinal: SnapshotOrdinal, state: CalculatedState)(implicit
+          context: L1NodeContext[F]
+        ): F[Boolean] = true.pure[F]
+
+        override def hashCalculatedState(state: CalculatedState)(implicit context: L1NodeContext[F]): F[Hash] =
+          Hash.empty.pure[F]
+
+        override def validateData(
+          state:   DataState[OnChain, CalculatedState],
+          updates: NonEmptyList[Signed[TodoUpdate]]
+        )(implicit context: L1NodeContext[F]): F[DataApplicationValidationErrorOr[Unit]] =
+          ().validNec[DataApplicationValidationError].pure[F]
+
+        override def validateUpdate(
+          update: TodoUpdate
+        )(implicit context: L1NodeContext[F]): F[DataApplicationValidationErrorOr[Unit]] =
+          context.getOnChainState.flatMap {
+            _.fold(
+              err => err.invalidNec[Unit].pure[F],
+              onchain => validator.verify(onchain, update)
+            )
+          }
+
+        override def combine(
+          state:   DataState[OnChain, CalculatedState],
+          updates: List[Signed[TodoUpdate]]
+        )(implicit context: L1NodeContext[F]): F[DataState[OnChain, CalculatedState]] = state.pure[F]
```

Comment:
does it work if you don't apply updates onto the previous state?

Reply:
The DataL1 never invokes the combine function but the inheritance from Tessellation is such that an implementation must be provided so this is just the simplest implementation.

Reply:
yeah, according to the previous comment, you shouldn't make separate implementations for each layer, but you should provide a single implementation and reuse it for each layer. so that, you don't need any dummy implementations

Reply:
I don't understand how would be possible to provide a single implementation for resuse between the Data-L1 and L0 services? Tessellation makes the distinction of `BaseDataApplicationL0Service` and `BaseDataApplicationL1Service` where the `Context` type constraint from `BaseDataApplicationService` is concretely tied to `L0NodeContext` and `L1NodeContext`. I agree that functions within each service could share functional definitions by pointing to some object as done in [Dor](https://github.com/Constellation-Labs/dor-metagraph/blob/d85e808d12bbb491fce8e789fe72f2d736cc4328/metagraph/modules/data_l1/src/main/scala/com/my/dor_metagraph/data_l1/Main.scala#L49). But in the specific case of the `combine(...)(implicit L1NodeContext)` this function is not called within Tessellation but is required by inheritance.

It would be my preference to not include these extraneous methods within the definition of each service as well but from what I can see in Tessellation, that seems like a significant refactor of the Data Application inheritance structure and overall usage.